### DISCUSSION TITLE
Quickly switch to any window using FZF

### DISCUSSION THREAD

**Discussion by gmayerw** [xmonad/xmonad#490] [General]
I know xmonad has a [feature](https://hackage.haskell.org/package/xmonad-contrib-0.17.1/docs/XMonad-Actions-Navigation2D.html) where it pulls up squares of all open windows. What if I wanted to do something similar and pipe all open windows to an [FZF](https://github.com/junegunn/fzf) terminal instance and then use FZF to select the window.

Two questions:
1. How would I get the list of all open windows on every workspace?
2. How would I send the result to xmonad so it knows where to switch?

**Comment by liskin** [comment]
You could probably use wmctrl for both, as long as your xmonad is configured to support EWMH. Or you can just use https://github.com/davatorium/rofi which is kind of what you want, but its fuzzy matching algo is a tiny bit different than fzf.

**Reply by gmayerw** [reply]
Thanks. Would I just configure it here somewhere?

```
import XMonad hiding ((|||))
import XMonad.Actions.CycleWS
import XMonad.Hooks.EwmhDesktops
import XMonad.Layout.LayoutCombinators
import XMonad.Util.EZConfig (additionalKeysP)
import XMonad.Util.WorkspaceCompare
import qualified XMonad.StackSet as W

import XMonad.Hooks.EwmhDesktops

main :: IO ()
main = do
    xmonad $ ewmh def
        { modMask = mod4Mask
        , terminal = "~/.scripts/clear_vifm"
        , borderWidth = 0
        , layoutHook = Full ||| tiled
        } `additionalKeysP` myKeys
```

**Reply by gmayerw** [reply]
I updated the above code to be a working ewmh config. I got the fzf code working too using `wmctrl` like you said:

```
wmctrl -a "$(wmctrl -l | awk '{$1=$2=$3=""; print substr($0,4)}' | fzf)"
```

It's great!! Thank you! Case closed!

**Comment by portnov** [comment]
You can also try using X.P.Window in combination with X.P.FuzzyMatch.
